src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaFinalDecisionDisplayProvider.java

```diff
@@ -0,0 +1,76 @@
+package uk.gov.hmcts.reform.iacaseapi.domain.handlers.presubmit;
+
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCaseFieldDefinition.*;
+
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCase;
+
+
+@Component
+public class FtpaFinalDecisionDisplayProvider {
+
+    private final String[][] ftpaDisplayMatrix = {
```

Comment:
It is not bad, just very C++ style. Refactoring proposition:
```
        Map<Pair, String> decisionMap = new ImmutableMap.Builder<Pair, String>()
            .put(new ImmutablePair<>("firstDecision", "secondDecision"), "outcome")
            .build();
```

Reply:
Thanks for that, appreciate the comment. Will look to refactor this.

Reply:
Refactored to use ImmutablePair instead of 2D array.